### PR TITLE
Translate S3 user-defined metadata prefix to/from GCS custom metadata prefix

### PR DESCRIPTION
**Pull request by wd256**
## Description

This PR updates the functions which deal with the conversion from the GCS clients ObjectAttrs struct to Minios ObjectInfo.

Metadata which uses the S3 user-defined metadata prefix of "x-amz-meta" will now be translated to the use the GCS custom metadata prefix of "x-goog-meta" before being sent to GCS and vice versa when received.

## Motivation and Context

The change is to ensure that the user-defined metadata prefix specified by Google is used when an object is uploaded to GCS. It also brings the GCS gateway implementation inline with the OSS and Azure implementations which perform the same sort of translation.

## How Has This Been Tested?

Unit tests have been added and manual testing carried out.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway/gcs/gateway-gcs.go [hunk 1]
```diff
@@ -799,21 +804,25 @@ func fromGCSAttrsToObjectInfo(attrs *storage.ObjectAttrs) minio.ObjectInfo {
...
-	}
-	// Filter metadata which is stored as a unique attribute
-	for _, key := range []string{
-		"content-type", "content-encoding", "cache-control", "content-disposition", "content-language",
-	} {
-		delete(attrs.Metadata, key)
+		k = http.CanonicalHeaderKey(k)
+		switch {
+		case strings.HasPrefix(k, "X-Amz-Meta-"):
+			// Translate the S3 used-defined metadata prefix
```
Comment (krisis) [inline cmd/gateway/gcs/gateway-gcs.go]: minor nit (typo): s/used-defined/user-defined

**Review by krisis — COMMENTED** [review]
The PR looks good to me. I will approve it once the typo is fixed. Thank you for fixing this issue.

File: cmd/gateway/gcs/gateway-gcs.go (hunk 1 above)
Comment (wd256) [reply cmd/gateway/gcs/gateway-gcs.go]: Cheers, typo is now resolved

**Review by krisis — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by nitisht** [comment]
@wd256 I need to get in touch with you, could you please drop me a line at nitish@minio.io

**Pull request merged**